File: src/main/java/com/indiepost/service/AdminPostService.java
package com.indiepost.service;

import com.indiepost.dto.post.AdminPostRequestDto;
import com.indiepost.dto.post.AdminPostResponseDto;
import com.indiepost.dto.post.AdminPostSummaryDto;
import com.indiepost.dto.post.PostQuery;
import com.indiepost.enums.Types;
import com.indiepost.model.Post;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

/**
 * Created by jake on 17. 1. 14.
 */
public interface AdminPostService {

    AdminPostResponseDto findOne(Long id);

    Long createDraft(AdminPostRequestDto dto);

    Long createAutosave(AdminPostRequestDto requestDto);

    void update(AdminPostRequestDto postUpdateDto);

    void updateAutosave(AdminPostRequestDto requestDto);

    Long deleteById(Long id);

    Long delete(Post post);

    Page<AdminPostSummaryDto> find(Types.PostStatus status, Pageable pageable);

    Page<AdminPostSummaryDto> findIdsIn(List<Long> ids, Pageable pageable);

    Page<AdminPostSummaryDto> findText(String text, Types.PostStatus status, Pageable pageable);

    Page<AdminPostSummaryDto> fullTextSearch(String text, Types.PostStatus status,
                                             Pageable pageable);

    Long count();

    Long count(PostQuery query);

    List<String> findAllBylineNames();

    void bulkDeleteByStatus(Types.PostStatus status);

    void bulkDeleteByIds(List<Long> ids);

    void bulkStatusUpdate(List<Long> ids, Types.PostStatus changeTo);

}

File: src/main/kotlin/com/indiepost/service/AdminPostService.kt
package com.indiepost.service

import com.indiepost.dto.post.*
import com.indiepost.enums.Types
import com.indiepost.model.Post
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

/**
 * Created by jake on 17. 1. 14.
 */
interface AdminPostService {

    fun getAllTitles(): List<Title>

    fun findOne(id: Long): AdminPostResponseDto?

    fun createDraft(dto: AdminPostRequestDto): Long?

    fun createAutosave(requestDto: AdminPostRequestDto): Long?

    fun update(requestDto: AdminPostRequestDto)

    fun updateAutosave(requestDto: AdminPostRequestDto)

    fun deleteById(id: Long): Long?

    fun delete(post: Post): Long?

    fun find(status: Types.PostStatus, pageable: Pageable): Page<AdminPostSummaryDto>

    fun findIdsIn(ids: List<Long>, pageable: Pageable): Page<AdminPostSummaryDto>

    fun findText(text: String, status: Types.PostStatus, pageable: Pageable): Page<AdminPostSummaryDto>

    fun fullTextSearch(text: String, status: Types.PostStatus,
                       pageable: Pageable): Page<AdminPostSummaryDto>

    fun count(): Long

    fun count(query: PostQuery): Long

    fun findAllBylineNames(): List<String>

    fun bulkDeleteByStatus(status: Types.PostStatus)

    fun bulkDeleteByIds(ids: List<Long>)

    fun bulkStatusUpdate(ids: List<Long>, changeTo: Types.PostStatus)

}
